src/test/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStepTest.java

```diff
@@ -177,6 +178,24 @@ public void milestoneNotAllowedInsideParallel() {
         });
     }
 
+    @Issue("JENKINS-38464")
+    @Test
```

Comment:
Sure. I was not going to merge this, just writing the test and leaving the fix for later.
@svanoort is finishing it.